### PR TITLE
Properly escape otp-runner manifest file

### PR DESCRIPTION
**Pull request by evansiroky**
### Checklist

- [x] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
- [x] Any modified or new methods or classes have helpful JavaDoc and code is thoroughly commented
- [x] The description lists all applicable issues this PR seeks to resolve
- [x] The description lists any configuration setting(s) that differ from the default settings
- [x] All tests and CI builds passing

### Description

This PR fixes the generation of the user data for ec2 instances so that the otp-runner manifest is properly written with properly escaped characters.

### CODE REVIEW

File: src/test/java/com/conveyal/datatools/manager/jobs/DeployJobTest.java [hunk 1]
```diff
@@ -17,22 +18,26 @@
...
 import static org.junit.Assert.assertFalse;
+import static org.junit.Assume.assumeTrue;
 
 /**
  * This contains a few helpful tests for quickly spinning up a deployment using any bundle of GTFS + OSM or pre-built
  * graph that already exists on S3.
  *
- * Note: these tests require credentials on the IBI AWS account, which is why the class is tagged with Ignore (so that it is
- * not run with the rest of the test suite)
+ * Note: some tests require credentials on the IBI AWS account, which is why those tests contain an assumeTrue statement
```
Comment (landonreed) [inline src/test/java/com/conveyal/datatools/manager/jobs/DeployJobTest.java:37]: Can this comment be modified slightly to suggest how to run those tests (e.g., set `RUN_AWS_DEPLOY_JOB_TESTS=true` to run these tests).

File: src/test/java/com/conveyal/datatools/manager/jobs/DeployJobTest.java [hunk 2]
```diff
@@ -63,19 +68,55 @@ public static void setUp() throws IOException {
         server.ec2Info.iamInstanceProfileArn = "YOUR_VALUE";
         Persistence.servers.create(server);
         deployment = new Deployment();
+        // the feedVersionIds don't get populated during test, so we have to manually do it here
+        deployment.feedVersionIds = new ArrayList<>();
```
Comment (landonreed) [inline src/test/java/com/conveyal/datatools/manager/jobs/DeployJobTest.java:72]: Are any IDs actually "populated"? I don't immediately see where that happens.

File: src/test/java/com/conveyal/datatools/manager/jobs/DeployJobTest.java [hunk 3]
```diff
@@ -63,19 +68,55 @@ public static void setUp() throws IOException {
...
+            "test-deploy",
+            DeployJob.DeployType.REPLACE
+        );
+        assertThat(deployJob.constructUserData(true), matchesSnapshot());
+    }
+
     /**
      * Tests that Data Tools can run an ELB deployment.
      *
      * Note: this requires changing server.yml#modules.deployment.ec2.enabled to true
```
Comment (landonreed) [inline src/test/java/com/conveyal/datatools/manager/jobs/DeployJobTest.java:115]: Does this need to be updated to match the class-level javadoc?

**Comment by evansiroky** [comment]
More comments have been added to the DeployJobTest.java class.

**Pull request merged**